improve: add contributing guidelines (#117)
across-protocol/across-frontend: README.md

- REACT_APP_DEFAULT_GAS_PRICE: default gas price estimate when no online estimate is available, specify in gwei
- REACT_APP_UPDATE_GAS_INTERVAL_MS: how frequently to update gas prices in MS, used for estimating adding liquidity when maxing eth. Default 30000 (30 sec).

> ## Contributing Best Practices

Reviewer: Some of these apply _only_ to our internal team. It's a little confusing to add things like shortcut into a public set of contribution guidelines. Do you think it makes sense to have a label like [Internal team only] to denote the points that don't apply to external contributors?

Author: Im going to specifically direct this doc to internal contributors since we currently do not have a process for external contributors

- All changes to master should first be a pull request.
- PR titles should conform to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).
- For those who have access to make branches, its recommended to branch directly from the repo for easier collaboration.
- Branch names should ideally be generated by [Shortcut](www.shortcut.com) issues, but otherwise theres no particular naming convention.
- Try to keep PR Focused on a single feature, bug or issue. 
- Document motivation for change in the PR, ie link to issue or describe reason for change
- If the change is relatively complex, document how you accomplished the change 
- If you want to make multiple changes in a PR, make sure they are small and document the motivation for each.
- Include screenshots when applicable. Before and after is ideal, though after is ok. 
- Tag at least 2 reviewers. Most PRs should be merged in after 2 approvals, but low risk or high urgency PRs can be merged with 1.

> - The owner of the PR merges the PR once aproved

Author: im going to direct this towards internal contributors only

> ### Feature Branches 

Author: this section is a sub section of the previous

Use feature branches when you need to split up changes to the app which may break the app or introduce incomplete features.
Name the branch based on the epic, story, feature or issue, which ideally links back to the original shortcut card.
Denote a feature branch with the prefix "epic/", for example "epic/l1tol2". We will reserve "epic" for feature branches.

#### Maintaing Branches

> For all branches which may be long lived, it may generally get out of date with your upstream branch. Its recommended to rebase your branch

Reviewer: ```suggestion
To avoid long-lived branches diverging from `master`, it's recommended to rebase
```

rather than merge changes from master. This makes it easy to spot problems during review as the commit history should only show your changes.

> Frequency of rebasing is dependent on how quickly your branch is getting out of date, but usually you want to do it soon after you see changes on your upstream branch.

Reviewer: ```suggestion
The frequency of rebasing is dependent on how quickly your branch is getting out of date, but you should generally rebase soon after you see changes on your upstream branch.
```

> Its also recommended to rebase before opening a PR.

Reviewer: ```suggestion
It's also recommended to rebase before opening a PR.
```

#### Quick Rebase Guide

> 1. Be on your branch. Ensure there are no outstanding changes, if so commit or stash them. 

Reviewer: ```suggestion
1. Start on your branch. Ensure there are no uncommitted changes, if so commit or stash them. 
```

2. `git fetch origin master` - This will fetch latest changes to master without switching branches.

> 3. `git rebase HEAD~X --onto origin/master` - Rebase your changes on top of the latest changes on master, where X is the number of commits you have made to your branch.

Reviewer: Note: this only works if the user's origin is the main across repo. This will _probably_ not be the case for forkers.

Not sure what the standard setup for a forker is, but I have generally labeled the main repo as "upstream". Regardless, you might want to note that origin should be replaced with whatever the label is for the across repo (not their fork).

Author: I've updated the doc to say it assumes internal contributors who have branched, since thats what we want best practices to be. I've also made another note to say we are assuming the upstream repo is `origin`
